### PR TITLE
snap telemetry 0.17.0

### PR DESCRIPTION
**Pull request by nanliu**
- [x] Have you followed the guidelines in our [Contributing](https://github.com/Homebrew/homebrew-core/blob/master/CONTRIBUTING.md) document?
- [x] Have you checked to ensure there aren't other open [Pull Requests](https://github.com/Homebrew/homebrew-core/pulls) for the same formula update/change?
- [x] Have you built your formula locally prior to submission with `brew install <formula>` (where `<formula>` is the name of the formula you're submitting)?
- [x] Does your submission pass `brew audit --new-formula <formula>` (after doing `brew install <formula>`)?

---

### CODE REVIEW

**Comment by ilovezfs** [comment]
@BrewTestBot test this please

File: Formula/snap-telemetry.rb [hunk 1]
```diff
@@ -0,0 +1,38 @@
+class SnapTelemetry < Formula
+  desc "Snap is an opensource telemetry framework"
+  homepage "http://snap-telemetry.io/"
+  url "https://github.com/intelsdi-x/snap/archive/v0.16.1-beta.tar.gz"
```
Comment (MikeMcQuaid) [inline Formula/snap-telemetry.rb]: This needs to be a stable version before we can accept it (i.e. not `beta`).

File: Formula/snap-telemetry.rb [hunk 2]
```diff
@@ -0,0 +1,38 @@
...
+    snapctl = buildpath/"src/github.com/intelsdi-x/snap/cmd/snapctl"
+    cd snapctl do
+      system "go", "build", "-o", "snapctl", "-ldflags", "-w -X main.gitversion=#{version}"
+      bin.install "snapctl"
+    end
+  end
+
+  test do
+    assert_match version.to_s, shell_output("#{bin}/snapd --version")
+    assert_match version.to_s, shell_output("#{bin}/snapctl --version")
```
Comment (MikeMcQuaid) [inline Formula/snap-telemetry.rb]: Can this test be modified to do something more substantial than e.g. `--version` or `--help`? See `cmake.rb` for an example of an application formula with a good test and `tinyxml2.rb` for an example of a library formula with a good test. Thanks!

**Review by MikeMcQuaid — CHANGES_REQUESTED** [review]

File: Formula/snap-telemetry.rb (hunk 1 above)
Comment (MikeMcQuaid) [inline Formula/snap-telemetry.rb]: Yeh, let's hold off for a non-beta e.g. 1.0.0.

File: Formula/snap-telemetry.rb (hunk 2 above)
Comment (nanliu) [inline Formula/snap-telemetry.rb]: I've added some additional tests below.

**Comment by nanliu** [comment]
This is updated to version 0.17.0, please review again.

**Comment by nanliu** [comment]
Err, not sure what the test step is hanging on jenkins. Locally it runs fine:

```
$ brew test snap-telemetry --verbose
git config --file=/usr/local/Homebrew/.git/config --replace-all homebrew.devcmdrun true
Testing snap-telemetry
==> Using the sandbox
/usr/bin/sandbox-exec -f /tmp/homebrew20161021-7563-qbkmp8.sb /System/Library/Frameworks/Ruby.framework/Versions/2.0/usr/bin/ruby -W0 -I /usr/local/Homebrew/Library/Homebrew -- /usr/local/Homebrew/Library/Homebrew/test.rb /usr/local/Homebrew/Library/Taps/homebrew/homebrew-core/Formula/snap-telemetry.rb --verbose
==> /usr/local/Cellar/snap-telemetry/0.17.0/bin/snapd --version
==> /usr/local/Cellar/snap-telemetry/0.17.0/bin/snapctl --version
==> /usr/local/Cellar/snap-telemetry/0.17.0/bin/snapctl plugin list
==> /usr/local/Cellar/snap-telemetry/0.17.0/bin/snapctl task list
==> kill 7580
```

File: Formula/snap-telemetry.rb [hunk 3]
```diff
@@ -0,0 +1,49 @@
...
+    snapd_pid = fork do
+      exec "#{bin}/snapd -t 0 -l 1 -o /tmp"
+    end
+
```
Comment (MikeMcQuaid) [inline Formula/snap-telemetry.rb]: May need a `sleep` here.

File: Formula/snap-telemetry.rb [hunk 4]
```diff
@@ -0,0 +1,49 @@
...
+    assert_match /No plugins/, shell_output("#{bin}/snapctl plugin list")
+    assert_match /No task/, shell_output("#{bin}/snapctl task list")
+
+    begin
+      system "kill", snapd_pid
+    ensure
+      Process.kill("TERM", snapd_pid)
```
Comment (MikeMcQuaid) [inline Formula/snap-telemetry.rb]: Just use `Process.kill` and put the `fork` inside this `begin` to ensure it'll always be killed if the `assert` fails.

**Comment by nanliu** [comment]
Yep, seems like it needed a short sleep for the daemon to startup. It's passing Jenkins successfully now.

**Comment by nanliu** [comment]
Ping? Please review again.

**Comment by candysmurf** [comment]
@MikeMcQuaid, if you have time to review this PR? Tomorrow is the last day of our sprint. It would be nice it's merged. thanks!

**Review by MikeMcQuaid — APPROVED** [review]

**Comment by MikeMcQuaid** [comment]
@BrewTestBot test this please

**Comment by MikeMcQuaid** [comment]
Thanks for your contribution to Homebrew! Without people like you submitting PRs we couldn't run this project. You rock!

**Pull request closed without merging**